nav2_map_server/include/nav2_map_server/map_server.hpp

```diff
@@ -24,10 +24,12 @@
 #include "nav_msgs/msg/occupancy_grid.hpp"
 #include "nav_msgs/srv/get_map.hpp"
 #include "nav2_msgs/srv/load_map.hpp"
```

Comment:
The map server should be either-or 2D or 3D. If someone has a 2D map and a 3D map, they can launch 2 instances of the map server to deal with that. Its a pretty typical workflow to have multiple map servers for each map source.

Reply:
That's exactly what it is designed for, as soon as Yaml inputs are available ad the node gets configured after that if someone tries to change it from 2D to 3D or opposite via using another YAML(of a different type than that of configured one), an error will welcome him.

Reply:
It looks like this is trying to support 2D and 3D rather than 2D or 3D

Reply:
Please have a look at https://github.com/ros-planning/navigation2/pull/1898/files#diff-7e36958d0fa1934277369188be334f47, the core of what it does is in map_server.cpp file

Reply:
shared_ptr won't be adding much overhead to memory that's why they are there as so.

Reply:
https://github.com/ros-planning/navigation2/pull/1898#discussion_r470508183